<!DOCTYPE html><html><head><title>R: Print Call Stacks</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body> <div style='padding: 20pt; border: 2px solid black; text-align:center;'><b>This help topic is for R version 2.2.1. For the corresponding topic in the current version of R, see <a href='https://stat.ethz.ch/R-manual/R-patched/library/base/html/traceback.html'>https://stat.ethz.ch/R-manual/R-patched/library/base/html/traceback.html</a></b></div><div class="container">

<table style="width: 100%;"><tr><td>traceback {base}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id='traceback'>Print Call Stacks</h2>

<h3>Description</h3>

<p>By default, <code>traceback()</code> prints the call stack of the last
uncaught error, i.e., the sequence of calls that lead to the error.
This is useful when an error occurs with an unidentifiable error
message.  It can also be used to print arbitrary lists of deparsed
calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traceback(x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traceback_:_x">x</code></td>
<td>
<p><code>NULL</code> (default), or a list or pairlist of deparsed
calls.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The stack of the last uncaught error is stored as a list of deparsed
calls in <code>.Traceback</code>, which <code>traceback</code> prints in a
user-friendly format.
</p>
<p>Errors which are caught <em>via</em> <code>try</code> or
<code>tryCatch</code> do not generate a traceback, so what is printed
is the call sequence for the last uncaught error, and not necessarily
the last error.
</p>


<h3>Value</h3>

<p><code>traceback()</code> returns nothing, but prints the deparsed call stack
deepest call first.  The calls may print on more that one line, and
the first line is labelled by the frame number.
</p>


<h3>References</h3>

<p>Becker, R. A., Chambers, J. M. and Wilks, A. R. (1988)
<em>The New S Language</em>.
Wadsworth \&amp; Brooks/Cole.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>foo &lt;- function(x) { print(1); bar(2) }
bar &lt;- function(x) { x + a.variable.which.does.not.exist }
## Not run: 
foo(2) # gives a strange error
traceback()
## End(Not run)
## 2: bar(2)
## 1: foo(2)
bar
## Ah, this is the culprit ...
</code></pre>

<hr /><div style="text-align: center;">[<a href='/r-help/2.2.1/00index.html'>Package <em>base</em> version 2.2.1</a> ]</div>
</div>
</body></html>
